Source/WebCore/Modules/pdfjs-extras/content-script.js

```diff
@@ -56,10 +56,19 @@ const PDFJSContentScript = {
     },
 
     init() {
-        this.overrideSettings();
-        PDFViewerApplication.eventBus.on("pagesinit", () => {
-            this.autoResize();
-        });
+        // FIXME: should initialize at the correct time instead of relying on a
+        // 200ms timeout loop. But I can't figure out how to make it work.
+        // https://github.com/mozilla/pdf.js/wiki/Third-party-viewer-usage#initialization-promise
+        // https://bugs.webkit.org/show_bug.cgi?id=251887
+        let id = setInterval(() => {
+            if (PDFViewerApplication.initialized) {
+                this.overrideSettings();
+                PDFViewerApplication.eventBus.on("pagesinit", () => {
+                    this.autoResize();
+                });
+                clearInterval(id);
+            }
+        }, 200);
```

Comment:
Can we do:

```js
             PDFViewerApplication.initializedPromise.then(function() {
                 this.overrideSettings();
                 PDFViewerApplication.eventBus.on("pagesinit", () => {
                     this.autoResize();
                 });
             });
```
?

Reply:
Ah, almost! If I change it to use an arrow function so that `this` gets captured, then yes that works. Will do. Nice.

Reply:
Done, please re-review.